example/oven_fsm.dart

```diff
@@ -0,0 +1,252 @@
+            led < LEDLight.blue.value,
+            counterReset < 1,
+            en < 0,
+          ]),
+
+      // identifier: cooking state, represent by `OvenState.cooking`.
+      State<OvenState>(OvenState.cooking,
+          // events:
+          // When the button `paused` is pressed during cooking state,
+          // OvenState will changed to `OvenState.paused` state.
+          //
+          // When the button `counter` time is elapsed during cooking state,
+          // OvenState will changed to `OvenState.completed` state.
+          events: {
+            Logic(name: 'button_pause')
+                  ..gets(button
+                      .eq(Const(Button.pause.value, width: button.width))):
+                OvenState.paused,
+            Logic(name: 'counter_time_complete')..gets(counter.val.eq(4)):
+                OvenState.completed
+          },
+          // actions:
+          // During the cooking state, `led` is change to yellow; timer's
+          // `counterReset` is set to 0 (Do not reset);
+          // timer's `en` is set to 1 (Enable value update).
+          actions: [
+            led < LEDLight.yellow.value,
+            counterReset < 0,
+            en < 1,
+          ]),
+
+      // identifier: paused state, represent by `OvenState.paused`.
+      State<OvenState>(OvenState.paused,
+          // events:
+          // When the button `resume` is pressed during paused state,
+          // OvenState will changed to `OvenState.cooking` state.
+          events: {
+            Logic(name: 'button_resume')
+                  ..gets(button
+                      .eq(Const(Button.resume.value, width: button.width))):
+                OvenState.cooking
+          },
+          // actions:
+          // During the paused state, `led` is change to red; timer's
+          // `counterReset` is set to 0 (Do not reset);
+          // timer's `en` is set to 0 (Disable value update).
+          actions: [
+            led < LEDLight.red.value,
+            counterReset < 0,
+            en < 0,
+          ]),
+
+      // identifier: completed state, represent by `OvenState.completed`.
+      State<OvenState>(OvenState.completed,
+          // events:
+          // When the button `start` is pressed during completed state,
+          // OvenState will changed to `OvenState.cooking` state.
+          events: {
+            Logic(name: 'button_start')
+                  ..gets(button
+                      .eq(Const(Button.start.value, width: button.width))):
+                OvenState.cooking
+          },
+          // actions:
+          // During the start state, `led` is change to green; timer's
+          // `counterReset` is set to 1 (Reset value);
+          // timer's `en` is set to 0 (Disable value update).
+          actions: [
+            led < LEDLight.green.value,
+            counterReset < 1,
+            en < 0,
+          ])
+    ];
+
+    // Assign the _oven StateMachine object to public variable declared.
```

Comment:
Sorry, I didn't explain correctly. I wanted to point out that perhaps the `public variable declared` part should be updated as the variable is now private. Are there any problems in the `oven StateMachine` part?

Reply:
I see what you mean now. Should be changed to private variable declared instead.